src/transformers/models/wav2vec2_bert/configuration_wav2vec2_bert.py

```diff
@@ -311,4 +309,7 @@ def __init__(
 
     @property
     def inputs_to_logits_ratio(self):
-        return functools.reduce(operator.mul, self.conv_stride, 1)
+        ratio = self.feature_projection_input_dim * 2
+        if self.add_adapter:
+            ratio = ratio * (self.adapter_stride**self.num_adapter_layers)
+        return ratio
```

Comment:
For reference, I've double-checked the ratio.


W2V2-Bert uses [M4T feature extractor](https://github.com/huggingface/transformers/blob/0e75aeefaf4beeb7a5bb6a1f05b83ab99e045a24/src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py#L38), which reproduces kaldi mel-spectrogram computation.
To compute the ratio:
1. Get ratio from mel-spectrogram. It can be found [here](https://github.com/pytorch/audio/blob/02586da797dbfa201721d2080e2171805202f72c/src/torchaudio/compliance/kaldi.py#L67). It is basically `1 + (waveform_length - frame_length) // hop_length`.
2. Divide by [`stride`](https://github.com/huggingface/transformers/blob/0e75aeefaf4beeb7a5bb6a1f05b83ab99e045a24/src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py#L292-L294).
3. Multiply by the model downsample factor -> it appears if an adapter block has been added on top of the model

Here, for the two first steps, `frame_length=400`, `hop_length=160` and `stride=2`. Exact computation gives `logits_length= waveform_length / 320 - 0.75`.
I'll round the ratio to `320` (`hop_length*ratio`). We can get an approximation of the ratio thanks to [`config.feature_projection_input_dim`](https://github.com/huggingface/transformers/blob/0e75aeefaf4beeb7a5bb6a1f05b83ab99e045a24/src/transformers/models/wav2vec2_bert/configuration_wav2vec2_bert.py#L57C9-L57C37) which is computed like this `(hop_length//2) * ratio`.

So the ratio of the two first steps should be `feature_projection_input_dim*2`.

Reply:
Great! Thanks for the clear explanation!